File: Challenger-Api/src/main/java/com/salesianostriana/dam/challengerapi/reserva/dto/NewReservaDTO.java
package com.salesianostriana.dam.challengerapi.reserva.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.salesianostriana.dam.challengerapi.reserva.model.Reserva;
import com.salesianostriana.dam.challengerapi.reserva.model.TipoDispositivo;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDateTime;

public record NewReservaDTO(

        @NotNull
        TipoDispositivo tipoDispositivo,

        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd/MM/yyyy HH:mm")
        LocalDateTime desde,

        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd/MM/yyyy HH:mm")
        LocalDateTime hasta
) { }

File: Challenger-Mobile/lib/models/dto/new_reserva_dto.dart
class NewReservaDto{
  String? tipoDispositivo;
  String? desde;
  String? hasta;

  NewReservaDto({this.tipoDispositivo, this.desde, this.hasta});

  NewReservaDto.fromJson(Map<String, dynamic> json) {
    tipoDispositivo = json['tipoDispositivo'];
    desde = json['desde'];
    hasta = json['hasta'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['tipoDispositivo'] = tipoDispositivo;
    data['desde'] = desde;
    data['hasta'] = hasta;
    return data;
  }

}